src/Compilers/Core/Portable/InternalUtilities/OneOrMany.cs

```diff
@@ -89,19 +89,13 @@ public OneOrMany<T> Add(T one)
 
         public bool Contains(T item)
         {
-            RoslynDebug.Assert(item != null);
             if (HasOne)
-            {
-                return item.Equals(_one);
-            }
+                return EqualityComparer<T>.Default.Equals(item, _one);
```

Comment:
this is the preferred equality pattern for generics.  The previous call would allocate, calling through object.Equals(object) if we had a struct.  teh new form will will specialize calling through `IEquatable<T>` in a non-allocating fashion if the struct implements that.